godot-core/src/init/mod.rs

```diff
@@ -119,6 +119,73 @@ unsafe extern "C" fn ffi_deinitialize_layer<E: ExtensionLibrary>(
     });
 }
 
+#[cfg(since_api = "4.3")]
+unsafe fn register() {
+    use crate::registry::plugin::PluginItem;
+    use std::collections::{hash_map::Entry, HashMap};
+    #[derive(Default)]
+    struct DocPieces {
+        /// base, description, members
+        definition: [&'static str; 3],
+        /// methods, signals, constants
+        inherent: [&'static str; 3],
+        imethods: &'static str,
+    }
+
+    let mut map = HashMap::<&'static str, DocPieces>::new();
+    crate::private::iterate_plugins(|x| match x.item {
+        PluginItem::InherentImpl { docs, .. } => match map.entry(x.class_name.as_str()) {
+            Entry::Occupied(x) => x.into_mut().inherent = docs,
+            Entry::Vacant(x) => drop(x.insert(DocPieces {
+                inherent: docs,
+                ..Default::default()
+            })),
+        },
+        PluginItem::ITraitImpl { docs, .. } => match map.entry(x.class_name.as_str()) {
+            Entry::Occupied(x) => x.into_mut().imethods = docs,
+            Entry::Vacant(x) => drop(x.insert(DocPieces {
+                imethods: docs,
+                ..Default::default()
+            })),
+        },
+        PluginItem::Struct { docs, .. } => match map.entry(x.class_name.as_str()) {
+            Entry::Occupied(x) => x.into_mut().definition = docs,
+            Entry::Vacant(x) => drop(x.insert(DocPieces {
+                definition: docs,
+                ..Default::default()
+            })),
+        },
+    });
+    for (
+        class,
+        DocPieces {
+            definition: [base, desc, members],
+            inherent: [methods, signals, constants],
+            imethods,
+        },
+    ) in map
+    {
+        let brief = desc.lines().next().unwrap_or_default();
+        let xml = format!(
+            r#"
+<?xml version="1.0" encoding="UTF-8"?>
+<class name="{class}" inherits="{base}" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+<brief_description>{brief}</brief_description>
+<description>{desc}</description>
+<methods>{methods}{imethods}</methods>
+<constants>{constants}</constants>
+<signals>{signals}</signals>
+<members>{members}</members>
+</class>"#
+        );
+        // SAFETY: the godot binding is initialized
+        crate::sys::interface_fn!(editor_help_load_xml_from_utf8_chars_and_len)(
+            xml.as_ptr().cast(),
+            xml.len() as _,
```

Comment:
please use an explicit type here, although you can probably use `try_into().expect()` instead since if this conversion fails then that's likely UB and it's likely not gonna be a very hot path.

Reply:
do we run on 16bit machines?

Reply:
the type its being cast to is `i64`, from a `String::len` which has a max size of `isize::MAX / 2`, and `i64::MAX > isize::MAX / 2`, so it doesnt need a try_into().

Reply:
assuming that godot wont ever support targets with bigger pointer width. which is probably a fair assumption but it'd be nice to explicitly state that somewhere. we can do that later though but it'd be nice to add a `// TODO` for it maybe. like `// TODO: add a usize -> i64 conversion in sys::conv`